### ISSUE TITLE
DenseSIFTDescriptor.get_pooling_kernel() hands out an alias of its buffer, captured before .to()

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4083]
`DenseSIFTDescriptor.get_pooling_kernel()` returns a tensor captured at construction time. It is an alias of the module's buffer rather than a copy, and it is captured *before* any `.to()`, so it both lets a caller corrupt the module and goes stale on device/dtype migration.

`kornia/feature/siftdesc.py`:

```python
# Cache pooling kernel torch.Tensor for fast return in get_pooling_kernel
self._pooling_kernel = self._bin_pooling_kernel_weight.detach()

def get_pooling_kernel(self) -> torch.Tensor:
    # Return the cached detached pooling kernel directly for optimal speed
    return self._pooling_kernel
```

`.detach()` shares storage, so `self._pooling_kernel` is a permanent view of the `_bin_pooling_kernel_weight` buffer as it existed in `__init__`.

## Two symptoms

**1. A caller's in-place write corrupts the module.**

```python
import torch
from kornia.feature import DenseSIFTDescriptor

m = DenseSIFTDescriptor()
m.get_pooling_kernel().zero_()
print(bool(m._bin_pooling_kernel_weight.abs().sum() == 0))
```
```
True
```

The accessor is documented as returning a "detached tensor", which reads as safe to use, and this is the same shape of defect as #4068 — just instance-local rather than process-global, so milder.

**2. The alias goes stale on `.to()`.** `nn.Module._apply` replaces the buffer with a new tensor; the Python attribute still points at the old one.

```python
import torch
from kornia.feature import DenseSIFTDescriptor

m = DenseSIFTDescriptor().to(torch.float64)
print(m._bin_pooling_kernel_weight.dtype, m.get_pooling_kernel().dtype)
```
```
torch.float64 torch.float32
```

The same happens for `.to("cuda")` or `.to("mps")`: the accessor keeps returning the CPU tensor for the lifetime of the module. This is the more likely one to bite, since `.to(device)` is routine and nothing raises.

## Proposed fix

Drop `self._pooling_kernel` and return `self._bin_pooling_kernel_weight.detach().clone()` (or at minimum re-read the buffer on each call). The cache saves nothing measurable — the accessor is not on any hot path, and `forward` does not use it.

`SIFTDescriptor.get_pooling_kernel` / `get_weighting_kernel` return `self.pk.weight.detach()` / `self.gk.detach()` fresh on each call, so they track `.to()` correctly; they still hand out an alias, so symptom 1 applies there too and is worth fixing in the same pass.

Reproduced on `main` at 6fa75d64 with torch 2.9.1, CPU.

Raised out of the review of #4074, which fixes #4068 and deliberately leaves this alone as a separate defect.
